azure-pipelines.yml

```diff
@@ -175,31 +175,32 @@ stages:
             # the unusual install method. For the bug report about missing .pc
             # files: https://github.com/MacPython/openblas-libs/issues/74
             set -euo pipefail
-            docker pull i386/ubuntu:bionic
-            docker run -v $(pwd):/scipy i386/ubuntu:bionic /usr/bin/linux32 /bin/bash -c "cd scipy && \
+            docker pull i386/ubuntu:focal
+            docker run -v $(pwd):/scipy i386/ubuntu:focal /usr/bin/linux32 /bin/bash -c "cd scipy && \
+            export DEBIAN_FRONTEND=nonintercative && \
             apt-get -y update && \
-            apt-get -y install curl python3.8-dev python3.8 python3-distutils pkg-config libpng-dev libjpeg8-dev libfreetype6-dev && \
+            apt-get -y install curl python3.9-dev python3.9 python3-distutils pkg-config libpng-dev libjpeg8-dev libfreetype6-dev && \
             curl https://bootstrap.pypa.io/get-pip.py -o get-pip.py && \
-            python3.8 get-pip.py && \
+            python3.9 get-pip.py && \
             pip3 --version && \
-            pip3 install setuptools==59.6.0 wheel numpy==1.19.5 cython==0.29.21 pybind11 pytest pytest-timeout pytest-xdist pytest-env pytest-cov Pillow mpmath pythran pooch && \
+            pip3 install setuptools==59.6.0 wheel numpy==1.21.6 cython==0.29.21 pybind11 pytest pytest-timeout pytest-xdist pytest-env pytest-cov Pillow mpmath pythran pooch && \
             apt-get -y install gcc-8 g++-8 gfortran-8 wget && \
             cd .. && \
             mkdir openblas && cd openblas && \
-            target=\$(python3.8 ../scipy/tools/openblas_support.py) && \
+            target=\$(python3.9 ../scipy/tools/openblas_support.py) && \
             cp -r \$target/lib/* /usr/lib && \
             cp \$target/include/* /usr/include && \
             cd ../scipy && \
-            CC=gcc-8 CXX=g++-8 F77=gfortran-8 F90=gfortran-8 python3.8 setup.py install && \
-            python3.8 tools/openblas_support.py --check_version $(openblas_version) && \
-            python3.8 runtests.py -n --mode=full -- -n auto -s --junitxml=junit/test-results.xml --cov-config=.coveragerc --cov-report=xml --cov-report=html --durations=10 --timeout=60"
+            CC=gcc-8 CXX=g++-8 F77=gfortran-8 F90=gfortran-8 python3.9 setup.py install && \
+            python3.9 tools/openblas_support.py --check_version $(openblas_version) && \
+            python3.9 runtests.py -n --mode=full -- -n auto -s --junitxml=junit/test-results.xml --cov-config=.coveragerc --cov-report=xml --cov-report=html --durations=10 --timeout=60"
```

Comment:
I wonder why we chose the Linux 32 bit run (of all things) to generate the coverage reports.

Reply:
Plus I think we don't use this one since we have this job `prerelease_deps_coverage_64bit_blas` (which btw I had a PR to move to GH actions #15958 but stalled)

Reply:
No particular reason I think, just an accident of history. Can easily be moved.

Reply:
I would do that in a follow up to not delay this one more.